go-controller/pkg/ovn/egressip.go

```diff
@@ -1726,57 +1811,41 @@ func (e *egressIPZoneController) createReroutePolicyOps(ops []ovsdb.Operation, p
 // the LogicalRouterPolicy completely
 // This function should be called with lock on nodeZoneState cache key status.Node
-func (e *egressIPZoneController) deleteReroutePolicyOps(ops []ovsdb.Operation, podIPNets []*net.IPNet, status egressipv1.EgressIPStatusItem, egressIPName string) ([]ovsdb.Operation, error) {
+func (e *egressIPZoneController) deleteReroutePolicyOps(ops []ovsdb.Operation, podIPNets []*net.IPNet, status egressipv1.EgressIPStatusItem, egressIPName, nextHopIP string) ([]ovsdb.Operation, error) {
 	isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
-	var nextHopIP string
 	var err error
-	isLocalZoneEgressNode, loadedEgressNode := e.nodeZoneState.Load(status.Node)
-	if loadedEgressNode && isLocalZoneEgressNode {
-		gatewayRouterIP, err := e.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
-		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %w", status.Node, isEgressIPv6, err)
-		}
-		nextHopIP = gatewayRouterIP.String()
-	} else if config.OVNKubernetesFeature.EnableInterconnect {
-		// fetch node annotation of the egress node
-		nextHopIP, err = e.getTransitIP(status.Node, isEgressIPv6)
-		if apierrors.IsNotFound(err) {
-			// if the node is already gone, then don't count it as error as during deletion.
-			klog.Warningf("Unable to fetch transit switch IP for node: %s: err: %v", status.Node, err)
-		} else if err != nil {
-			return nil, fmt.Errorf("unable to fetch transit switch IP for node %s: %v", status.Node, err)
-		}
-	}
-	// if neither of these above conditions are met, return error since we can't remove an empty nextHop LRP
-	if nextHopIP == "" {
-		return nil, fmt.Errorf("unable to determine nextHop for egressIP %s with status %v", egressIPName, status)
-	}
 	// Handle all pod IPs that match the egress IP address family
 	for _, podIPNet := range util.MatchAllIPNetFamily(isEgressIPv6, podIPNets) {
 		filterOption := fmt.Sprintf("%s.src == %s", ipFamilyName(isEgressIPv6), podIPNet.IP.String())
 		p := func(item *nbdb.LogicalRouterPolicy) bool {
 			return item.Match == filterOption && item.Priority == types.EgressIPReroutePriority && item.ExternalIDs["name"] == egressIPName
 		}
-		ops, err = libovsdbops.DeleteNextHopFromLogicalRouterPoliciesWithPredicateOps(e.nbClient, ops, types.OVNClusterRouter, p, nextHopIP)
-		if err != nil {
-			return nil, fmt.Errorf("error removing nexthop IP %s from egress ip %s policies on router %s: %v",
-				nextHopIP, egressIPName, types.OVNClusterRouter, err)
+		if nextHopIP != "" {
+			ops, err = libovsdbops.DeleteNextHopFromLogicalRouterPoliciesWithPredicateOps(e.nbClient, ops, types.OVNClusterRouter, p, nextHopIP)
+			if err != nil {
+				return nil, fmt.Errorf("error removing nexthop IP %s from egress ip %s policies on router %s: %v",
+					nextHopIP, egressIPName, types.OVNClusterRouter, err)
+			}
+		} else {
+			// since next hop was not found, delete everything to ensure no stale entries however this will break load
```

Comment:
This will de-configure the whole `egressIP` object for a particular pod breaking any HA if there were multiple IPs defined.
I think we should log an error message here and update the doc string of the function

Reply:
Logged an error and updated the doc string.